lib/plsql/procedure_call.rb

```diff
@@ -236,6 +236,14 @@ def add_argument(argument, value, argument_metadata=nil)
         @bind_values[argument] = value.nil? ? nil : (value ? 1 : 0)
         @bind_metadata[argument] = argument_metadata.merge(:data_type => "NUMBER", :data_precision => 1)
         "l_#{argument}"
+      when 'UNDEFINED'
+        if argument_metadata[:type_name] == 'XMLTYPE'
+          @declare_sql << "l_#{argument} XMLTYPE;\n"
+          @assignment_sql << "l_#{argument} := XMLTYPE(:#{argument});\n" if not value.nil?
+          @bind_values[argument] = value if not value.nil?
+          @bind_metadata[argument] = argument_metadata.merge(:data_type => "CLOB")
+          "l_#{argument}"
+        end
```

Comment:
Hi Yavor.
I'll look into it this week.
Thanks for spotting that potential behavior change.
Jacek
On 25 Oct 2015 18:33, "Yavor Nikolov" <notifications@github.com> wrote:

> In lib/plsql/procedure_call.rb
> <https://github.com/rsim/ruby-plsql/pull/91#discussion_r42948271>:
>
> > @@ -236,6 +236,14 @@ def add_argument(argument, value, argument_metadata=nil)
> >          @bind_values[argument] = value.nil? ? nil : (value ? 1 : 0)
> >          @bind_metadata[argument] = argument_metadata.merge(:data_type => "NUMBER", :data_precision => 1)
> >          "l_#{argument}"
> > +      when 'UNDEFINED'
> > +        if argument_metadata[:type_name] == 'XMLTYPE'
> > +          @declare_sql << "l_#{argument} XMLTYPE;\n"
> > +          @assignment_sql << "l_#{argument} := XMLTYPE(:#{argument});\n" if not value.nil?
> > +          @bind_values[argument] = value if not value.nil?
> > +          @bind_metadata[argument] = argument_metadata.merge(:data_type => "CLOB")
> > +          "l_#{argument}"
> > +        end
>
> @jgebal <https://github.com/jgebal>, any opinion about that question? My
> concern is that we're maybe changing the semantics of non-xmltype
> 'UNDEFINED' (even though I'm not sure what exactly that might be).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rsim/ruby-plsql/pull/91/files#r42948271>.
>